src/node_v8.cc

```diff
@@ -243,6 +249,66 @@ void SetFlagsFromString(const FunctionCallbackInfo<Value>& args) {
   V8::SetFlagsFromString(*flags, static_cast<size_t>(flags.length()));
 }
 
+class JSONOutputStream : public v8::OutputStream {
+  public:
+   JSONOutputStream() {}
+
+   int GetChunkSize() override { return 65536; }
+
+   void EndOfStream() override {}
+
+   WriteResult WriteAsciiChunk(char* data, const int size) override {
+     out_stream_.write(data, size);
+     return kContinue;
+   }
+
+   std::ostringstream& out_stream() { return out_stream_; }
+
+  private:
+   std::ostringstream out_stream_;
+};
+
+void StartCpuProfile(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  CHECK(args[0]->IsString());
+  Isolate* isolate = env->isolate();
+  String::Utf8Value name(isolate, args[0]);
+  Local<String> title = String::NewFromUtf8(isolate,
+                                            *name,
+                                            NewStringType::kNormal,
+                                            name.length())
+                                            .ToLocalChecked();
```

Comment:
```suggestion
  Local<String> title;
  if (!String::NewFromUtf8(isolate,
                           *name,
                           NewStringType::kNormal,
                           name.length())
                           .ToLocal(&title)) {
    // An error will have been scheduled by v8
    return;
  }
```

Reply:
> An error will have been scheduled by v8

Yeah, the problem is ... this is wrong. 😢 https://github.com/v8/v8/blob/071d3af83fd318f60453cadf2f610c5217e25f3a/src/api/api.cc#L7578

It is really unfortunate that V8 is pretty consistent about what `Maybe<>` and `MaybeLocal<>` indicate, _except_ in this one particular case.

So if we care about proper error handling (I think we do!), then the best way to do that is by using Node.js's `ToV8Value()`, that is generally better for a number of reasons, including already being able to handle standard C++ types and the consistent error handling.

Reply:
Updated the suggest to use `ToV8Value(...)` :-)